ophyd/controls/device.py

```diff
@@ -0,0 +1,439 @@
+import time
+import inspect
+from collections import (OrderedDict, namedtuple)
+
+from .ophydobj import (OphydObject, DeviceStatus)
+from ..utils import TimeoutError
+
+
+class Component:
+    '''A descriptor representing a device component (or signal)
+
+    Unrecognized keyword arguments will be passed directly to the component
+    class initializer.
+
+    Parameters
+    ----------
+    cls : class
+        Class of signal to create
+    suffix : str
+        The PV suffix, which gets appended onto the device prefix
+    add_prefix : sequence, optional
```

Comment:
I find the name confusing, and I wonder if there could be a more obvious implementation, such as `prefix_template='{suffix}{sep}{write_pv}'`. I'm willing to punt on this, though, if you like your way.